### PR TITLE
remove adal go package and disable ADX log route

### PR DESCRIPTION
**Pull request by pfrcks**
This pull request includes changes to remove the ADX (Azure Data Explorer) related code and dependencies from the project, as well as a change to the `.trivyignore` file to remove a critical telegraf vulnerability. The most important changes include removing the ADX related code from the `oms.go` and `utils.go` files, removing the ADX related dependencies from the `go.mod` file, and removing the critical telegraf vulnerability from the `.trivyignore` file.

Removal of ADX related code:

* [`source/plugins/go/src/oms.go`](diffhunk://#diff-b11de461d67b679d0a9f2fe929616504690bd41bf4e9cec27d7a5791933a171eL127-L129): Removed all the code related to ADX, including the creation of the ADX client, the routing of container logs, and the posting of data helper. [[1]](diffhunk://#diff-b11de461d67b679d0a9f2fe929616504690bd41bf4e9cec27d7a5791933a171eL127-L129) [[2]](diffhunk://#diff-b11de461d67b679d0a9f2fe929616504690bd41bf4e9cec27d7a5791933a171eL145-L146) [[3]](diffhunk://#diff-b11de461d67b679d0a9f2fe929616504690bd41bf4e9cec27d7a5791933a171eL189-L190) [[4]](diffhunk://#diff-b11de461d67b679d0a9f2fe929616504690bd41bf4e9cec27d7a5791933a171eL1842-R1833) [[5]](diffhunk://#diff-b11de461d67b679d0a9f2fe929616504690bd41bf4e9cec27d7a5791933a171eL2270-R2211) [[6]](diffhunk://#diff-b11de461d67b679d0a9f2fe929616504690bd41bf4e9cec27d7a5791933a171eL2358-L2359)
* [`source/plugins/go/src/utils.go`](diffhunk://#diff-de3771438420bafe6137baebb8599dd304fe27cc69d837d0a9cbe11bcbbf6abdL205-L229): Removed the `CreateADXClient` function which was used to create the ADX client.

Removal of ADX related dependencies:

* [`source/plugins/go/src/go.mod`](diffhunk://#diff-4c8eb2bbf5d4a6f8f0d84217e96f7fb96f5d919c8f8e2519e5f99e86cd3f9262L8): Removed all the dependencies related to Azure, including the Azure Kusto Go SDK, which was used for interaction with ADX. [[1]](diffhunk://#diff-4c8eb2bbf5d4a6f8f0d84217e96f7fb96f5d919c8f8e2519e5f99e86cd3f9262L8) [[2]](diffhunk://#diff-4c8eb2bbf5d4a6f8f0d84217e96f7fb96f5d919c8f8e2519e5f99e86cd3f9262L25-L39) [[3]](diffhunk://#diff-4c8eb2bbf5d4a6f8f0d84217e96f7fb96f5d919c8f8e2519e5f99e86cd3f9262L48-L66)
* [`source/plugins/go/src/oms.go`](diffhunk://#diff-b11de461d67b679d0a9f2fe929616504690bd41bf4e9cec27d7a5791933a171eL29): Removed the import of the Azure Kusto Go SDK.
* [`source/plugins/go/src/utils.go`](diffhunk://#diff-de3771438420bafe6137baebb8599dd304fe27cc69d837d0a9cbe11bcbbf6abdL18-L19): Removed the import of the Azure Kusto Go SDK.

Removal of critical telegraf vulnerability:

* [`.trivyignore`](diffhunk://#diff-a2b39041fe7520e8037b4e596d015ea4825043e383f3fa5f85d5ac7f06f4c177L13-L15): Removed the critical telegraf vulnerability `CVE-2024-24790`.

Other changes:

* [`build/common/installer/scripts/tomlparser.rb`](diffhunk://#diff-be1193a49b27f3fec5e115daf986f4479bcfd5a010fc3518e090a1a6180e3177L27): Removed the code related to the ADX database name setting. [[1]](diffhunk://#diff-be1193a49b27f3fec5e115daf986f4479bcfd5a010fc3518e090a1a6180e3177L27) [[2]](diffhunk://#diff-be1193a49b27f3fec5e115daf986f4479bcfd5a010fc3518e090a1a6180e3177L289-L304) [[3]](diffhunk://#diff-be1193a49b27f3fec5e115daf986f4479bcfd5a010fc3518e090a1a6180e3177L389) [[4]](diffhunk://#diff-be1193a49b27f3fec5e115daf986f4479bcfd5a010fc3518e090a1a6180e3177L459-L460)

### CODE REVIEW

File: .trivyignore [hunk 1]
```diff
@@ -10,6 +10,3 @@ CVE-2024-27304
 CVE-2023-45288
 CVE-2024-24786
 CVE-2024-24791
-
-#telegraf CRITICAL
```
Comment (ganga1980) [inline .trivyignore:14]: is this gone with adx code removal?

**Review by ganga1980 — APPROVED** [review]

**Review by ganga1980 — COMMENTED** [review]
changes lgtm. can you validate image with these changes and verify no crashes and logs flows as expected?

File: .trivyignore (hunk 1 above)
Comment (pfrcks) [reply .trivyignore:14]: this is gone because telegraf has been rebuilt to fix this vuln in the latest mariner which we are consuming

**Comment by pfrcks** [comment]
validated that logs are flowing and no crashes. ADX mode is now a NOOP and logs error in the fluent-bit log files

**Pull request merged**